book/src/concepts/system.md

```diff
@@ -0,0 +1,249 @@
+
+### Removing an entity
+
+Deleting an entity is very easy using `Entities<'a>`.
+```rust
+entities.delete(entity);
+```
+
+### Iterating over components with associated entity
+
+Sometimes, when you iterate over components, you may want to also know what entity you are working with. To do that, you can use the joining operation with `Entities<'a>`.
+
+```rust,ignore
+struct MakeObjectsFall;
+
+impl<'a> System<'a> for MakeObjectsFall {
+    type SystemData = (
+        Entities<'a>,
+        WriteStorage<'a, Transform>,
+        ReadStorage<'a, FallingObject>,
+    );
+
+    fn run(&mut self, (entities, mut transforms, falling): Self::SystemData) {
+        for (e, mut transform, _) in (&*entities, &mut transforms, &falling).join() {
+            if transform.translation.y > 0.0 {
+                transform.translation.y -= 0.1;
+            } else {
+                entities.delete(e);
+            }
+        }
+    }
+}
+```
+
+This system does the same thing as the previous `MakeObjectsFall`, but also cleans up falling objects that reached the ground.
+
+### Adding or removing components
+
+You can also insert or removes components from a specific entity.
+To do that, you need to get a mutable storage of the component you want to modify, and simply do:
+
+```rust,ignore
+// Add the component
+write_storage.insert(entity, MyComponent);
+
+// Remove the component
+write_storage.remove(entity);
+```
+
+Keep in mind that inserting a component on an entity that already has a component of the same type **will overwrite the previous one**.
+
+## The SystemData trait
+
+While this is a rarely useful, it is possible to create custom `SystemData` types.
+
+The `Dispatcher` populates the `SystemData` on every call of the `run` method. To do that, your `SystemData` type must implement the trait `amethyst::ecs::SystemData` in order to have it be valid.
+
+This is rather complicated trait to implement, fortunately Amethyst provides a derive macro for it, that can implement the trait to any struct as long as all its fields are `SystemData`. Most of the time however, you will not even need to implement it at all as you will be using `SystemData` structs provided by the engine.
+
+Please note that tuples of structs implementing `SystemData` are themselves `SystemData`. This is very useful when you need to request multiple `SystemData` at once quickly.
+
+```rust,ignore
+#[derive(SystemData)]
+struct MySystemData<'a> {
+    foo: ReadStorage<'a, FooComponent>,
+    bar: WriteStorage<'a, BarComponent>,
+    baz: BazSystemData<'a>,
+}
+
+struct MyFirstSystem;
+
+impl<'a> System<'a> for MyFirstSystem {
+    type SystemData = MySystemData<'a>;
+
+    fn run(&mut self, data: Self::SystemData) {
+        if data.baz.should_process() {
+            for (foo, mut bar) in (&data.foo, &mut data.bar) {
+                bar.stuff += foo.stuff;
+            } 
+        }
+    }
+}
+```
+
+## The setup method
```

Comment:
Wouldn't we want the setup method to be before the SystemData trait?
It feels more useful than that trait to me.